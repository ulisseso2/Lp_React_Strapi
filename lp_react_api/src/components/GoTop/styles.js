import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    background-color: red;
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    background-color: red;
  `}
`;
