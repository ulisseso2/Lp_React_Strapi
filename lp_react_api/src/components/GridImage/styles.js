import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css``}
`;

export const Title = styled.div`
  ${({ theme }) => css``}
`;
export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    align-items: center;
  `}
`;
export const GridImg = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
    margin: 10px;
  `}
`;

export const Img = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    transition: all 300ms ease-in-out;
    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
