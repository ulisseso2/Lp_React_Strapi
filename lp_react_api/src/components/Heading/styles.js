import styled from 'styled-components';
import { css } from 'styled-components';

export const Title = styled.h1`
  ${({ theme, colorDark }) => css`
    color: ${colorDark ? theme.colors.hot : theme.colors.dark};
  `};
`;
// styled component do componente Heading
