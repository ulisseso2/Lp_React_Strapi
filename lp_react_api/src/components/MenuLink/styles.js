import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    font-size: ${theme.font.sizes.small};
    text-decoration: none;
    padding: ${theme.spacings.small};
    color: ${theme.colors.promaryColor};
    font-weight: 400;
    position: relative;
    &::after {
      content: '';
      position: absolute;
      bottom: 0;
      left: 50%;
      height: 0.2rem;
      width: 0px;
      background-color: ${theme.colors.secondaryColor};
      transition: all 300ms ease-in-out;
    }
    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
