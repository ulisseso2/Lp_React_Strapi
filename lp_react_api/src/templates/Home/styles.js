import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.primary};
    color: ${theme.colors.hot};
    font-family: ${theme.fonts.main};
  `}
  padding: 10px;
`;
export const H1 = styled.h1`
  ${(props) => css`
    color: ${props.color};
    font-size: 30px;
  `}
`;
