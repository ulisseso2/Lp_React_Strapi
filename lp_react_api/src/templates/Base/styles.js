import styled, { css } from 'styled-components';

import { Container as TextComponent } from '../../components/Textcomponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
  `}
`;
