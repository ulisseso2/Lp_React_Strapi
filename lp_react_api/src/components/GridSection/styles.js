import styled, { css } from 'styled-components';

import { Title as HeadingContainer } from '../Heading/styles';
import { Container as Textcomponent } from '../Textcomponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${Textcomponent} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    overflow: hidden;
    width: 100%;

    @media ${theme.font.media.ltmedium} {
      grid-template-columns: 1fr;
    }
  `}
`;
export const GridElement = styled.div`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }
    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 6rem;
      font-weight: 900;
      left: -4rem;
      top: -1.5rem;
      transform: rotate(7deg);
    }

`;
