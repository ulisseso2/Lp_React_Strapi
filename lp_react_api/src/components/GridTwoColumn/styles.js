import styled, { css } from 'styled-components';

import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme, background }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};
    @media ${theme.font.media.ltmedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }
    ${Title} {
      margin-bottom: ${theme.spacings.xlarge};
      color: ${background ? theme.colors.white : theme.colors.primaryColor};
    }
  `}
`;
export const TextContainer = styled.div`
  ${({ theme }) => css``}
`;
export const ImgContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    max-width: 500px;
  `}
`;
export const Image = styled.img`
  ${({ theme }) => css`
    max-width: 500px;
  `}
`;
