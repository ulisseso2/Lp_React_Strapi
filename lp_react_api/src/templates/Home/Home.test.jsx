import { toHaveStyle } from '@testing-library/jest-dom/dist/matchers';
import { render, screen } from '@testing-library/react';
import { Home } from '.';
import { renderTheme } from '../../styles/render.theme';
import { theme } from '../../styles/theme';

test('render <Home/>', () => {
  const { degug } = renderTheme(<Home />);
  const headingContainer = screen.getByRole('heading', {
    name: 'Hello, World!',
  }).parentElement;
  expect(<Home />).toBeInTheDocument;
  expect(headingContainer).toHaveStyle({ background: theme.colors.primary });
  expect(headingContainer).toMatchSnapshot();
});
