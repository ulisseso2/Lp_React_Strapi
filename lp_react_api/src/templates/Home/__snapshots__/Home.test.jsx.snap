// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Home/> 1`] = `
<div
  class="sc-bczRLJ hlsKoZ"
>
  <h1
    class="sc-gsnTZi elfauH"
    color="red"
  >
    Hello, World!
  </h1>
  <p>
    ol√°
  </p>
  <input
    placeholder="Write a color in English"
    type="text"
  />
</div>
`;
